lec03-basic-r.Rmd

```diff
@@ -1,4 +1,748 @@
+function is contained in an R-package called
+[`readr`](http://readr.tidyverse.org/). R-packages are a bit like browser
+extensions, they are not essential, but can provide nifty functionality. We will
+go through R-packages in general and which ones are good for data analyses in
+in detail later in this lecture. Now, let's install `readr`:
+
+```{r, eval=FALSE}
+install.packages('readr') # You only need to install a package once
+```
+
+Now we can use the `read_csv` function. One useful option that `read_csv`
+includes, is the ability to read a CSV file directly from a URL, without
+downloading it in a separate step:
+
+```{r, eval=FALSE}
+surveys <- readr::read_csv('https://ndownloader.figshare.com/files/2292169')
+```
+
+However, it is often a good idea to download the data first, so you have a copy
+stored locally on your computer in case you want to do some offline analyses, or
+the online version of the file changes or the file is taken down. You can either
+download the data manually or from within R:
+
+```{r, eval=FALSE}
+download.file("https://ndownloader.figshare.com/files/2292169",
+              "portal_data.csv") # Saves this name in the current directory
+```
+
+The data is read in by specifying its local path.
+
+```{r}
+surveys <- readr::read_csv('portal_data.csv')
+```
+
+This statement produces some output regarding which data type it found in each 
+column. If we want to check this in more detail, we can print the variable's 
+value: `surveys`.
+
+```{r}
+surveys
+```
+
+In the online html-version of this lecture, you only see the first few rows of
+this data frame. Running this code in the R Notebook, would display a nice
+tabular view of the data, which also includes pagination when there are many
+rows and we can click this arrow to view all the columns. Technically, this is
+actually a `tibble` rather than a data frame as indicated in the output. This is
+because we used `read_csv` to load it. Since a `tibble` is just a data frame
+with some convenient extra functionality, we will use these words
+interchangeably from now on.
+
+If we just want to glance at how the data frame looks, it is sufficient to display only the
+top (the first 6 lines) using the function `head()`:
+
+```{r}
+head(surveys)
+```
+
+## What are data frames?
+
+Data frames are the _de facto_ data structure for most tabular data, and what we
+use for statistics and plotting. A data frame can be created by hand, but most
+commonly they are generated by the function `read_csv()`. In other words, when
+importing spreadsheets from your hard drive (or the web).
+
+A data frame is the representation of data in the format of a table where the
+columns are vectors that all have the same length. Because the columns are
+vectors, they all contain the same type of data as we discussed in last class
+(e.g., characters, integers, factors). We can see this when inspecting the
+**str**ucture of a data frame with the function `str()`:
```

Comment:
For me it looks like all text from here is bold in the GitHub pull request. It looks good locally and works to knit.

Reply:
Inconsistent markdown specifications...

Reply:
I see, I removed the bold for now. Although it would render fine on the home page, I don't think it is necessary.